Support for retriable and timeout configuration

aviscasillas opened the pull request:
This PR provides ActiveGraphQL with:
- Support of retry strategy based on [Retriable](https://github.com/kamui/retriable).
- Support of requests timeout configuration.

Having these changes in place, an ActiveGraphQL model could be configured doing something like:

``` ruby
class Model < ActiveGraphQL::Model
   configure url: 'http://some.api.domain',
             timeout: 0.1,
             retriable: { tries: 10,
                          base_interval: 0.5,
                          max_interval: 1 }
end
```

aviscasillas committed:
Add support for retriable and timeout

menno commented:
Why not simply pass the options given to `retriable` to the `Retriable.retriable` block? Maybe I want to use some options that you have not added to `retriable_supported_options` (like `multiplier`). 

The same could be argued for http options ( `http: { timeout: 0.1 }` to easily support all available options of the httparty gem such as `http_proxyaddr` or the authentication ones.

aviscasillas commented:
Yes @menno it makes sense, it will be much more flexible and understandable.
I will change it 👍 .

aviscasillas committed:
More flexible configuration for retriable and http

This makes configuration for retriable and http more flexible, accepting all possible options for them.

menno commented:
Great, looks good 👍
